v0/destinations/salesforce/transform.js

```diff
@@ -235,11 +235,33 @@ async function getSalesforceIdFromPayload(message, authorizationData) {
         leadObjectId = leadQueryResponse.data.searchRecords[0].Id;
       }
     }
+    if (leadObjectId) {
+      const convertedDetails = await axios.get(
+        `${authorizationData.instanceUrl}/services/data/v${SF_API_VERSION}/sobjects/Lead/${leadObjectId}?fields=IsConverted,ConvertedContactId,IsDeleted`,
```

Comment:
Why not request for these fields in the query on Ln 211?

Reply:
Saves us a network call.

Reply:
SF is not allowing to request these fields alongwith the query, the call fails.